cmd/allinone/app/entrypoint.go

```diff
@@ -0,0 +1,75 @@
+package app
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+	apiserverApp "kubesphere.io/devops/cmd/apiserver/app"
+	"kubesphere.io/devops/cmd/apiserver/app/options"
+	controllerApp "kubesphere.io/devops/cmd/controller/app"
+	controllerOpt "kubesphere.io/devops/cmd/controller/app/options"
+	"kubesphere.io/devops/pkg/config"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+func NewCommand() *cobra.Command {
+	return &cobra.Command{
+		Use: "ks-devops",
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			stopChain := signals.SetupSignalHandler()
+			go func(stopCh <-chan struct{}) {
+				if err := runControllerManager(stopCh); err != nil {
+					panic(err)
+				}
+			}(stopChain)
+			err = runAPIServer(stopChain)
+			return
+		},
+	}
+}
+
+func runAPIServer(stopCh <-chan struct{}) error {
+	s := options.NewServerRunOptions()
+
+	// Load configuration from file
+	conf, err := config.TryLoadFromDisk()
+	if err == nil {
+		s = &options.ServerRunOptions{
+			GenericServerRunOptions: s.GenericServerRunOptions,
+			Config:                  conf,
+		}
+	} else {
+		klog.Fatal("Failed to load configuration from disk", err)
```

Comment:
It's better to log as warning or break current thread.

Reply:
Actually, it's not true. It cannot continue without a configuration.